packages/@aws-cdk/aws-stepfunctions-tasks/lib/call-dynamodb.ts

```diff
@@ -0,0 +1,789 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Stack } from '@aws-cdk/core';
+import { getResourceArn } from './resource-arn-suffix';
+
+/**
+ * Determines the level of detail about provisioned throughput consumption that is returned.
+ */
+export enum ReturnConsumedCapacity {
+  /**
+   * The response includes the aggregate ConsumedCapacity for the operation,
+   * together with ConsumedCapacity for each table and secondary index that was accessed
+   */
+  INDEXES = 'INDEXES',
+
+  /**
+   * The response includes only the aggregate ConsumedCapacity for the operation.
+   */
+  TOTAL = 'TOTAL',
+
+  /**
+   * No ConsumedCapacity details are included in the response.
+   */
+  NONE = 'NONE'
+}
+
+/**
+ * Determines whether item collection metrics are returned.
+ */
+export enum ReturnItemCollectionMetrics {
+  /**
+   * If set to SIZE, the response includes statistics about item collections,
+   * if any, that were modified during the operation are returned in the response
+   */
+  SIZE = 'SIZE',
+
+  /**
+   * If set to NONE, no statistics are returned.
+   */
+  NONE = 'NONE'
+}
+
+/**
+ * Use ReturnValues if you want to get the item attributes as they appear before or after they are changed
+ */
+export enum ReturnValues {
+  /**
+   * Nothing is returned
+   */
+  NONE = 'NONE',
+
+  /**
+   * Returns all of the attributes of the item
+   */
+  ALL_OLD = 'ALL_OLD',
+
+  /**
+   * Returns only the updated attributes
+   */
+  UPDATED_OLD = 'UPDATED_OLD',
+
+  /**
+   * Returns all of the attributes of the item
+   */
+  ALL_NEW = 'ALL_NEW',
+
+  /**
+   * Returns only the updated attributes
+   */
+  UPDATED_NEW = 'UPDATED_NEW'
+}
+
+/**
+ * Map of string to AttributeValue
+ */
+export interface AttributeValueMap {
+  [key: string]: AttributeValue;
+}
+
+/**
+ * Class to generate AttributeValue
+ */
+export class AttributeValue {
+  private attributeValue: any = {};
+
+  /**
+   * Sets an attribute of type String. For example:  "S": "Hello"
+   */
+  public addS(value: string) {
```

Comment:
Should we instead name this and the other methods `withS()` instead?

Reply:
Yep, that sounds good.